packages/core/lib/console.js

```diff
@@ -262,7 +262,21 @@ class Console extends EventEmitter {
           path.join(this.options.contracts_build_directory, file),
           "utf8"
         );
-        jsonBlobs.push(JSON.parse(body));
+        const json = JSON.parse(body);
+        const metadata = JSON.parse(json.metadata);
+        // filter out Truffle's console.log. We don't want users to interact with in the REPL.
+        // user contracts named console.log will be imported, and a warning will be issued.
+        if (
+          !Object.keys(metadata.sources).some(source => {
+            return (
+              source === "truffle/console.sol" ||
+              source === "truffle/Console.sol"
+            );
+          }) &&
+          Object.keys(metadata.sources).length === 1
```

Comment:
This seems inverted...

```suggestion
          Object.keys(metadata.sources).length > 1
```

Reply:
Is there an optimization opportunity here? 

```
        const { sources } = JSON.parse(json.metadata);
        const sourcesLength = Object.keys(sources).length;
        if (sourcesLength > 1 || sourcesLength === 1 &&
          !Object.keys(sources).some(source => {
            return (
              source === "truffle/console.sol" ||
              source === "truffle/Console.sol"
            );
          })
        ) {
          jsonBlobs.push(json);
        }
```